addons/web/static/lib/py.js/lib/py.js

```diff
@@ -256,15 +256,20 @@ var py = {};
 
     py.tokenize = (function () {
         function group() { return '(' + Array.prototype.join.call(arguments, '|') + ')'; }
+        function maybeGroup() {return group(...arguments) + '?'; }
 
         var Whitespace = '[ \\f\\t]*';
 
         var Name = '[a-zA-Z_]\\w*';
 
-        var DecNumber = '\\d+(L|l)?';
+        var DecNumber = '\\d+' + maybeGroup('L','l');
         var IntNumber = DecNumber;
-        var PointFloat = group('\\d+\\.\\d*', '\\.\\d+');
-        var FloatNumber = PointFloat;
+
+        var FloatWithDecimalPart = "\\d+" + maybeGroup("\\.\\d*");
+        var FloatWithLeadingDot = '\\.\\d+';
+        var Exponent = '[eE][+-]?\\d+';
+        var FloatNumber = group(FloatWithLeadingDot, FloatWithDecimalPart) + maybeGroup(Exponent);
+
```

Comment:
Doesn't seem correct, `FloatWithDecimalPart` matches numbers with no decimal point and `Exponent` is optional so this will match an integer as a float. A float is either a point float where the exponent is optional, or a number without a decimal point and a required exponent. In practice it doesn't matter because we treat floats and integers uniformly but this is not guaranteed to keep being the case 

```suggestion
        var Exponent = '[eE][+-]?\\d+';
        var PointFloat = group(`\\d+\\.\\d*(${Exponent})?`, `\\.\\d+(${Exponent})?`);
        // Exponent not optional when no decimal point
        var FloatNumber = group(PointFloat, `\\d+${Exponent}`);
```

Same comment applies to the other tokenizer